data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/OpenSearchIndexPartitionCreationSupplier.java

```diff
@@ -82,10 +82,18 @@ private List<PartitionIdentifier> applyForOpenSearchClient(final Map<String, Obj
 
         LOG.debug("Found {} indices", indicesResponse.valueBody().size());
 
-        return indicesResponse.valueBody().stream()
-                .filter(osIndicesRecord -> shouldIndexBeProcessed(osIndicesRecord.index()))
-                .map(indexRecord -> PartitionIdentifier.builder().withPartitionKey(indexRecord.index()).build())
-                .collect(Collectors.toList());
+        final List<PartitionIdentifier> partitions = new java.util.ArrayList<>();
+        int matchCount = 0;
+        for (final var esIndicesRecord : indicesResponse.valueBody()) {
+            if (shouldIndexBeProcessed(esIndicesRecord.index())) {
+                partitions.add(PartitionIdentifier.builder().withPartitionKey(esIndicesRecord.index()).build());
+                matchCount++;
+            }
+        }
+        if (matchCount == 0) {
+            LOG.info(NOISY, "No indices matched the configured regex patterns after applying include/exclude filters");
```

Comment:
Should this be a warn log?

Reply:
I do lean towards agreeing with @dlvenable here. It's unlikely someone would configure a pipeline that has no existing indexes matching the pattern on startup.

Technically, they could have a new index get created that matches the pattern after the pipeline starts, but we would then process an empty or near empty index since it's a one-time migration via point in time. Either way, if someone intends that, then WARN can't hurt. It is far more likely that someone misconfigures the pattern.